core/common/src/ContourDisplay.ts

```diff
@@ -0,0 +1,296 @@
+  public equals(other: Contour): boolean {
+    if (!this.majorStyle.equals(other.majorStyle) || !this.minorStyle.equals(other.minorStyle) || this.minorInterval !== other.minorInterval  || this.majorIntervalCount !== other.majorIntervalCount || this.showGeometry !== other.showGeometry) {
+      return false;
+    }
+    return true;
+  }
+
+  /** Compares two contours.
+     * @param lhs First contour to compare
+     * @param rhs Second contour to compare
+     * @returns 0 if lhs is equivalent to rhs, a negative number if lhs compares less than rhs, or a positive number if lhs compares greater than rhs.
+     */
+  public static compare(lhs: Contour, rhs: Contour): number {
+    let diff = 0;
+    if ((diff = ContourStyle.compare(lhs.majorStyle, rhs.majorStyle)) !== 0)
+      return diff;
+    if ((diff = ContourStyle.compare(lhs.minorStyle, rhs.minorStyle)) !== 0)
+      return diff;
+    if ((diff = compareNumbers(lhs.minorInterval, rhs.minorInterval)) !== 0)
+      return diff;
+    if ((diff = compareNumbers(lhs.majorIntervalCount, rhs.majorIntervalCount)) !== 0)
+      return diff;
+    if ((diff = compareBooleans(lhs.showGeometry, rhs.showGeometry)) !== 0)
+      return diff;
+
+    return diff;
+  }
+
+  private constructor(json?: ContourProps) {
+    if (undefined === json) {
+      this.majorStyle = ContourStyle.fromJSON({ pixelWidth: 2 });
+      this.minorStyle = ContourStyle.fromJSON();
+      this.minorInterval = 1;
+      this.majorIntervalCount = 5;
+      this.showGeometry = true;
+    } else {
+      this.majorStyle = json.majorStyle ? ContourStyle.fromJSON(json.majorStyle) : ContourStyle.fromJSON({ pixelWidth: 2 });
+      this.minorStyle = json.minorStyle ? ContourStyle.fromJSON(json.minorStyle) : ContourStyle.fromJSON();
+      this.minorInterval = json.minorInterval ?? 1;
+      this.majorIntervalCount = json.majorIntervalCount ?? 5;
+      this.showGeometry = json.showGeometry ?? true;
+    }
+  }
+
+  public static fromJSON(json?: ContourProps) {
+    return json ? new Contour(json) : new Contour({});
+  }
+
+  public toJSON(): ContourProps {
+    const props: ContourProps = {};
+
+    if (!this.majorStyle.equals(ContourStyle.fromJSON({ pixelWidth: 2 })))
+      props.majorStyle = this.majorStyle.toJSON();
+
+    if (!this.minorStyle.equals(ContourStyle.fromJSON()))
+      props.minorStyle = this.minorStyle.toJSON();
+
+    if (1 !== this.minorInterval)
+      props.minorInterval = this.minorInterval;
+
+    if (5 !== this.majorIntervalCount)
+      props.majorIntervalCount = this.majorIntervalCount;
+
+    if (true !== this.showGeometry)
+      props.showGeometry = this.showGeometry;
+
+    return props;
+  }
+}
+
+/** JSON representation of the description of how contours should appear for a particular set of subcategories. */
+export interface ContourGroupProps {
+  /** See [[ContourGroup.contourDef]]. */
+  contourDef?: ContourProps;
+  /** See [[ContourGroup.subCategories]]. */
+  subCategories?: CompressedId64Set;
```

Comment:
These Ids will be stored in the JSON properties of DisplayStyle elements. You'll need to update `DisplayStyle.collectReferenceIds` and `DisplayStyle.onCloned` to correctly disclose and remap them.

Reply:
I pushed some code for this. I am not too familiar with this, so if you could let me know if it looks good I'd appreciate it.

Reply:
Prefer to use a unit test to prove you got it right.

Reply:
Do you happen to know if we have any existing unit tests that execute this code flow that we can take a look at to base this one?

Reply:
I would have to grep.

Reply:
[Right here](https://github.com/iTwin/itwinjs-core/blob/12362e64b682cbff804cbd7becd75003e522ab0b/core/backend/src/test/standalone/DisplayStyle.test.ts#L83).

Reply:
Thanks. I added a unit test for contour subCategory remapping in DisplayStyle.test.ts in the backend dir. It found a problem and I also pushed a fix for that.